packages/components/src/sandbox/index.native.js

```diff
@@ -239,6 +243,21 @@ function Sandbox( {
 		return '<!DOCTYPE html>' + renderToString( htmlDoc );
 	}
 
+	const getInjectedJavaScript = useCallback( () => {
+		// Allow parent to override the resize observers with prop.customJS (legacy support)
+		let injectedJS = customJS || observeAndResizeJS;
+
+		// Add any event listeners that were passed in.
+		Object.keys( onWindowEvents ).forEach( ( eventType ) => {
+			injectedJS += `
+				window.addEventListener( '${ eventType }', function( event ) {
+					window.ReactNativeWebView.postMessage( JSON.stringify( { type: '${ eventType }', ...event.data } ) );
```

Comment:
I considered delegating the event listener injection to the parent but I opted for this route to better hide the internal implantation of the `WebView` component. Specifically I am trying to hide the need to know about `window.ReactNativeWebView.postMessage`. Instead the listeners will receive an object that looks more like the original window event.

Reply:
I don't think the IIFE will give us much protection here. From what I understand the IIFF in `observeAndResizeJS` is to protect scope since it declares some variables. If we want protection we should use a try/catch but I'm not sure that is needed either. The closure might fail trying to stringify the event data but we should be guarded the same way if we were to wrap the `window.addEventListner` in a function.